scripts/registry-census/summarize_matrix.py

```
@@ -0,0 +1,232 @@
+        held = measured_pct >= floor
+        lines.append(
+            f'  {label:42s} {measured_pct:5.1f}%  (floor {floor:.0f}%)'
+            f'  {"OK" if held else "BREACH"}'
+        )
+        if not held:
+            breaches.append(f'{label} {measured_pct:.1f}% < {floor:.0f}%')
+
+    run_id = os.environ.get('MATRIX_RUN_ID', '')
+    prev_path = os.environ.get('MATRIX_PREV', '')
+    if prev_path and os.path.exists(prev_path):
+        prev = json.load(open(prev_path))
```

Ruff (0.16.1)

[warning] Use a context manager for opening files

(SIM115)